e2e-tests/portal/cypress/e2e/journeys/dashboard/dashboard-deals/deal-cancellation-enabled/dashboard-deals-filter-by-cancellation-status.spec.js

```diff
@@ -0,0 +1,79 @@
+const relative = require('../../../../relativeURL');
+const MOCK_USERS = require('../../../../../../../e2e-fixtures');
+const { dashboardDeals } = require('../../../../pages');
+const { dashboardFilters } = require('../../../../partials');
+const { BSS_DEAL_READY_FOR_CHECK, GEF_DEAL_DRAFT, BSS_DEAL_CANCELLED } = require('../../fixtures');
+
+const { BANK1_MAKER1, ADMIN } = MOCK_USERS;
+
+const filters = dashboardFilters;
+const statusCheckboxSelectors = dashboardDeals.filters.panel.form.status;
+
+context('Dashboard Deals Cancellation status filter - Deal cancellation enabled', () => {
+  before(() => {
+    cy.deleteGefApplications(ADMIN);
+    cy.deleteDeals(ADMIN);
+
+    cy.insertOneDeal(BSS_DEAL_READY_FOR_CHECK, BANK1_MAKER1);
+    cy.insertOneGefApplication(GEF_DEAL_DRAFT, BANK1_MAKER1);
+    cy.insertOneGefApplication(BSS_DEAL_CANCELLED, BANK1_MAKER1);
+  });
+
+  describe('Cancelled status filter', () => {
+    before(() => {
+      cy.login(BANK1_MAKER1);
+      dashboardDeals.visit();
+      cy.url().should('eq', relative('/dashboard/deals/0'));
+    });
+
+    beforeEach(() => {
+      cy.saveSession();
+      dashboardDeals.visit();
+
+      // toggle to show filters (hidden by default)
+      filters.showHideButton().click();
+    });
+
+    it('submits the filter and redirects to the dashboard', () => {
+      // apply filter
+      statusCheckboxSelectors.cancelled.checkbox().click();
+      filters.panel.form.applyFiltersButton().click();
+
+      cy.url().should('eq', relative('/dashboard/deals/0'));
+    });
+
+    it('renders checked checkbox', () => {
+      statusCheckboxSelectors.cancelled.checkbox().should('be.checked');
+    });
+
+    it('renders the applied filter in the `applied filters` section', () => {
+      filters.panel.selectedFilters.container().should('be.visible');
+      filters.panel.selectedFilters.list().should('be.visible');
+
+      const firstAppliedFilterHeading = filters.panel.selectedFilters.heading().first();
+
+      firstAppliedFilterHeading.should('be.visible');
+      cy.assertText(firstAppliedFilterHeading, 'Status');
+
+      const firstAppliedFilter = filters.panel.selectedFilters.listItem().first();
+
+      firstAppliedFilter.should('be.visible');
+      const expectedText = `Remove this filter ${TFM_DEAL_STAGE.CANCELLED}`;
+      cy.assertText(firstAppliedFilter, expectedText);
+    });
+
+    it('renders the applied filter in the `main container selected filters` section', () => {
+      dashboardDeals.filters.mainContainer.selectedFilters.statusCancelled().should('be.visible');
+
+      const expectedText = `Remove this filter ${TFM_DEAL_STAGE.CANCELLED}`;
+      dashboardDeals.filters.mainContainer.selectedFilters.statusCancelled().contains(expectedText);
+    });
+
+    it('renders only cancelled deals', () => {
+      dashboardDeals.rows().should('have.length', 1);
+
+      cy.assertText(dashboardDeals.row.status(BSS_DEAL_CANCELLED._id), TFM_DEAL_STAGE.CANCELLED);
```

Comment:
Could you add a test that no others are rendered? I think there's a `row.length` property somewhere

Reply:
On facilities you can do `dashboardFacilities.rows().should('have.length', EXPECTED_FACILITIES.length);`, hopefully theres equiv for deal

Reply:
That or just add a `blah should not exist` test

Reply:
There is already a `dashboardDeals.rows().should('have.length', 1);` which I think covers it!

Reply:
Have refactored anyway to make consistent with the other tests. Will mark as resolved!